### PR TITLE
datascience version 2.0 changes

### PR DESCRIPTION
**Pull request by shishirtewari**
Review comments incorporated for datascience and created datascience2.0 branch. Please review.

### CODE REVIEW

File: data_science/datascience.rb [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+  def chisquare_p
+    analyzer_data = {}
+    @cohorts.uniq(&:name).each do |row|
+      analyzer_data[row.name] =
+          {
+            total: cohort_count(row.name),
+            conversion: conversion_count(row.name)
+          }
+    end
```
Comment (zsyed1) [inline data_science/datascience.rb]: You can move lines 25 - 28 into a separate method say `cohort_totals` or something. Then you can replace lines 22 - 28 with `analyzer_data = @cohorts.uniq(&:name).map { |row| cohort_totals(row) }`.

File: data_science/spec/cohort_spec.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
+load 'cohort.rb'
+require 'spec_helper'
+require 'rspec'
+
+describe Cohort do
+  let(:cohort) do
+    Cohort.new('date' => '2014-03-20', 'cohort' => 'B', 'result' => 0)
+  end
+  context ' when check object' do
+    it 'check date' do
```
Comment (zsyed1) [inline data_science/spec/cohort_spec.rb]: Extra space after the single-quote in line 9

File: data_science/datascience.rb [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+  def conversion_rate(name)
+    (conversion_count(name).to_f / cohort_count(name).to_f).round(4)
+  end
+
+  def leader
+    leader = []
+    @cohorts.uniq(&:name).each do |row|
+      leader << { name: row.name, rate: conversion_rate(row.name) }
+    end
+    leader.max_by { |_k, v| v }[:name]
```
Comment (zsyed1) [inline data_science/datascience.rb]: You could do a similar refactor here in lines 39 - 41, but I prefer the way you did it - you kept the method down to 5 lines total which is good!

File: data_science/spec/spec_helper.rb [hunk 4]
```diff
@@ -0,0 +1,3 @@
+require_relative '../fileparser.rb'
+require_relative '../cohort.rb'
+require_relative '../datascience.rb'
```
Comment (zsyed1) [inline data_science/spec/spec_helper.rb]: @djkotowski I think this is closed! :-)

File: data_science/spec/fileparser_spec.rb [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+    it 'Not a correct file' do
+      expect { FileParser.parse_data('test.json') }.to raise_error(RuntimeError)
+    end
+    it 'Not a valid arguments' do
+      expect { FileParser.parse_data }.to raise_error(ArgumentError)
+    end
+    it 'Good file' do
+      expect { FileParser.parse_data('sample.json') }.not_to raise_error
+    end
+  end
```
Comment (djkotowski) [inline data_science/spec/fileparser_spec.rb]: I might include a little more detail, ensuring that the parser returns the JSON in the expected format.  Minor detail though.

**Comment by djkotowski** [comment]
This looks good.  Take a look at the comments, but no need for resubmission.  You're done with this.

**Pull request closed without merging**